src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -942,3 +700,222 @@ def handle_member_approval(self, slack_request: SlackActionRequest, action: str)
+                    sample_rate=1.0,
+                    tags={"type": "update_modal"},
+                )
+                logging_data = slack_request.get_logging_data()
+                _logger.exception(
+                    "slack.action.update-modal-not-found",
+                    extra={
+                        **logging_data,
+                        "trigger_id": slack_request.data["trigger_id"],
+                        "dialog": self.dialog_type,
+                    },
+                )
+                # The modal was not found, so we need to open a new one
+                self._open_view(slack_client, modal_payload, slack_request)
+            else:
+                raise
+
+    def _open_view(
+        self,
+        slack_client: SlackSdkClient,
+        modal_payload: View,
+        slack_request: SlackActionRequest,
+    ) -> None:
+        # Error handling is done in the calling function
+        slack_client.views_open(
+            trigger_id=slack_request.data["trigger_id"],
+            view=modal_payload,
+        )
+
+    def open_dialog(self, slack_request: SlackActionRequest, group: Group) -> None:
+        # XXX(epurkhiser): In order to update the original message we have to
+        # keep track of the response_url in the callback_id. Definitely hacky,
+        # but seems like there's no other solutions [1]:
+        #
+        # [1]: https://stackoverflow.com/questions/46629852/update-a-bot-message-after-responding-to-a-slack-dialog#comment80795670_46629852
+        org = group.project.organization
+
+        callback_id_dict = {
+            "issue": group.id,
+            "orig_response_url": slack_request.data["response_url"],
+            "is_message": _is_message(slack_request.data),
+            "rule": slack_request.callback_data.get("rule"),
+        }
+
+        if slack_request.data.get("channel"):
+            callback_id_dict["channel_id"] = slack_request.data["channel"]["id"]
+        callback_id = orjson.dumps(callback_id_dict).decode()
+
+        # only add tags to metadata
+        metadata_dict = callback_id_dict.copy()
+        metadata_dict["tags"] = list(slack_request.get_tags())
+        metadata = orjson.dumps(metadata_dict).decode()
+
+        # XXX(CEO): the second you make a selection (without hitting Submit) it sends a slightly different request
+        modal_payload = self.get_modal_payload(callback_id, metadata=metadata)
+        slack_client = SlackSdkClient(integration_id=slack_request.integration.id)
+        try:
+            # We need to use the action_ts as the external_id to update the modal
+            # We passed this in control when we sent the loading modal to beat the 3 second timeout
+            external_id = slack_request.get_action_ts()
+
+            if options.get("send-slack-response-from-control-silo"):
+                self._update_modal(slack_client, external_id, modal_payload, slack_request)
```

Comment:
@iamrajjoshi is this option true by default now?

Reply:
yes, still need to cleanup the FF